Frends.Instagram.Request/Frends.Instagram.Request/Frends.Instagram.Request.cs

```diff
@@ -0,0 +1,154 @@
+    {
+        // Check if Content-Type exists and is set and valid
+        var contentTypeIsSetAndValid = false;
+        MediaTypeWithQualityHeaderValue validContentType = null;
+        if (headers.TryGetValue("content-type", out string contentTypeValue))
+            contentTypeIsSetAndValid = MediaTypeWithQualityHeaderValue.TryParse(contentTypeValue, out validContentType);
+
+        return contentTypeIsSetAndValid
+            ? new StringContent(input.Message ?? string.Empty, Encoding.GetEncoding(validContentType.CharSet ?? Encoding.UTF8.WebName))
+            : new StringContent(input.Message ?? string.Empty);
+    }
+
+    private static async Task<HttpResponseMessage> GetHttpRequestResponseAsync(
+        HttpClient httpClient,
+        string method,
+        string url,
+        HttpContent content,
+        IDictionary<string, string> headers,
+        CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        // Only POST, PUT, PATCH and DELETE can have content, otherwise the HttpClient will fail
+        var isContentAllowed = Enum.TryParse(method, ignoreCase: true, result: out SendMethods _);
+
+        using var request = new HttpRequestMessage(new HttpMethod(method), new Uri(url))
+        {
+            Content = isContentAllowed ? content : null,
+        };
+
+        // Clear default headers
+        content.Headers.Clear();
+        foreach (var header in headers)
+        {
+            var requestHeaderAddedSuccessfully = request.Headers.TryAddWithoutValidation(header.Key, header.Value);
+            if (!requestHeaderAddedSuccessfully && request.Content != null)
+            {
+                // Could not add to request headers try to add to content headers
+                // this check is probably not needed anymore as the new HttpClient does not seem fail on malformed headers
+                var contentHeaderAddedSuccessfully = content.Headers.TryAddWithoutValidation(header.Key, header.Value);
+                if (!contentHeaderAddedSuccessfully)
+                {
+                    Trace.TraceWarning($"Could not add header {header.Key}:{header.Value}");
+                }
+            }
+        }
+
+        HttpResponseMessage response;
+        try
+        {
+            response = await httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+        }
+        catch (TaskCanceledException canceledException)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                // Cancellation is from outside -> Just throw
+                throw;
+            }
+
+            // Cancellation is from inside of the request, mostly likely a timeout
+            throw new Exception("HttpRequest was canceled, most likely due to a timeout.", canceledException);
+        }
+
+        request.Dispose();
+        httpClient.Dispose();
```

Comment:
Please wrap these dispose() method calls inside a finally block so that they will be handled even if exception is thrown.